test/test/nonjava/common_non_java.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "common_non_java.h"
+
+asprof_init_t CommonNonJava::asprof_init;
+asprof_execute_t CommonNonJava::asprof_execute;
+asprof_error_str_t CommonNonJava::asprof_error_str;
+
+JavaVM* CommonNonJava::jvm;
+JNIEnv* CommonNonJava::env;
+
+void* CommonNonJava::jvmLib;
+
+void CommonNonJava::outputCallback(const char* buffer, size_t size) {
+    fwrite(buffer, sizeof(char), size, stderr);
+}
+
+void CommonNonJava::loadProfiler() {  
+    #ifdef __linux__
```

Comment:
IIRC, this won't compile with some older gcc. There should not be whilespace before `#`.
Btw, inline #ifdefs in a method implementation are discouraged anyway.